### PR TITLE
Fix invalid memory access that happens in some cases

### PR DESCRIPTION
**Pull request by horgh**

### CODE REVIEW

File: bin/mmdblookup.c [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #include <unistd.h>
 #endif
 
-#define LOCAL
+#define LOCAL static
```
Comment (oschwald) [inline bin/mmdblookup.c]: I wonder if we should just get rid of `LOCAL` everywhere. I find it more confusing than helpful.

File: src/maxminddb.c [hunk 2]
```diff
@@ -1740,36 +1771,47 @@ LOCAL int get_entry_data_list(MMDB_s *mmdb,
 
             offset = entry_data_list->entry_data.offset_to_next;
             while (size-- > 0) {
-                MMDB_entry_data_list_s *entry_data_list_to =
-                    data_pool_get(pool);
-                if (NULL == entry_data_list_to) {
+                size_t index2 = 0;
```
Comment (oschwald) [inline src/maxminddb.c]: The naming is a bit hard to keep track of, but it isn't a big deal. Maybe `key_index` and `value_index` or something would help.

File: bin/mmdblookup.c (hunk 1 above)
Comment (horgh) [reply bin/mmdblookup.c]: I'm on board with that. Using `static` seems clear to me. Not sure if `LOCAL` has any advantages.

File: src/maxminddb.c (hunk 2 above)
Comment (horgh) [reply src/maxminddb.c]: Good point. I'll change that.

**Pull request merged**